### DISCUSSION TITLE
Date range in github search code

### DISCUSSION THREAD

**Discussion by david26694** [community/community#56097] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I would like to use date ranges when using github search for code. For instance, this returns all pandas imports in github:

https://github.com/search?q=%22import+pandas+as+pd%22&type=code

I'd like to return all pandas imports in github with either of the two filters:
* File last committed in a certain date range
* File first committed in a certain date range

Is this in the roadmap?

**Comment by look** [comment, marked as answer]
Thanks for your question @david26694. Our new code search backend does not support date filtering. In order to prioritize this, it'd be useful to hear more about your use case for wanting to find all the pandas code on GitHub within certain date ranges.

**Reply by fulldecent** [reply]
I would like to search recent code. Today I want to find all PrestaShop modules (i.e. code that contains `_PS_MODULE_DIR_`). But many results are years old. I would like to filter last_committed within the past year. (Either that code was most recently committed, or the repo was most recently committed.)

**Reply by LiteBrite82** [reply]
In all of GitHub, or, a specific repo?

**Reply by fulldecent** [reply]
All GitHub. Using it to discover projects

**Reply by LiteBrite82** [reply]
We cant filter code by date. Maybe we could try coming at it from a different angle? 
Im not familiar with PrestaShop; do they have any proprietary file names? something unique to their code base?

**Reply by LiteBrite82** [reply]
my typical goto solution for this type of search is search the commit history. Though that depends on how detailed the commit messages are (the commit search only searches for terms in the commit message and authors etc..)

**Reply by LiteBrite82** [reply]
example: https://github.com/search?q=_PS_MODULE_DIR_&type=commits&s=committer-date&o=desc

**Reply by fulldecent** [reply]
Yes, this order is good

**Reply by Xiaoven** [reply]
I hope we can access the date information in the code search results. I want to find code before the launch of ChatGPT to avoid AI-generated code.

**Comment by david26694** [comment]
I'd like to use github search to monitor usage of a library I developed. Currently I do this:
https://github.com/search?q=%22from+cluster_experiments%22&type=code
and I get the total number of imports. However, I can't monitor number of files using it since last week or last month.

**Reply by look** [reply]
Thanks for following up. I'll let the team know your feedback.

**Reply by bsfjqfea** [reply]
Fuck u and fuck ur team lol

**Reply by bsfjqfea** [reply]
Very bad code search, many users will leave ur shitty plateform

**Reply by modcruz** [reply]
> Thanks for following up. I'll let the team know your feedback.

Hi @look , Any update on this requirement??

**Comment by ryboe** [comment]
We're debugging a very strange issue that's hard to reproduce. The only thing we know for certain from our logs is that it started at a certain time two days ago. It would be awesome if we could search for all commits in our org from around that time.

Here's what I'm imagining:

```
org:mycompany.com after:2024-04-21T09:00:00 before:2024-04-21T08:00:00
```

Thanks for considering this feature.

**Reply by look** [reply]
Commit search features committer date and author date (Git records both dates) filtering, [check out the documentation](https://docs.github.com/en/search-github/searching-on-github/searching-commits#search-by-authored-or-committed-date).

You example would be something like this:

```
org:mycompany committer-date:2024-04-21..2024-04-22
```

**Comment by kevinpeno** [comment]
Please consider adding support for the [Duration/Period](https://en.wikipedia.org/wiki/ISO_8601#Durations) syntax of ISO 8601. This would allow saved searches regardless of the current date. I stumbled upon this limitation today due to wanted to search for "help wanted" issues/PRs, but not ones so stale that it wasn't worth looking into. I'm sure I could come up with a number of everyday use cases for organizational code as well with enough effort.

**Example use cases:**

_Last updated greater than one year ago today:_

- Current: `updated:>2023-10-13` (must update date every single day, can save, but not really useful)
- Expected: `updated:>-P1Y` (implies "greater than one year ago from today", will always work if saved)

_Bob implemented that fix within the last month, I'm sure of it_

- Current: `user:bob author-date:>2024-09-13`
- Expected: `user:bob author-date:>-P1M`

_What changed in the last 24 hours to cause this outage_
- Current: `author-date:>2024-10-12` (must manually create each time)
- Expected: `author-date:>-P1D` (can save for fire drills)

_Complex use case for saved PR/issue search_ (technically what I tried to accomplish today)

-  Current: `is:public state:open label:help-wanted,helpwanted,help+wanted,help,hacktoberfest updated:>2023-10-13` (could save all except updated since portions and need to change daily)
- Expected: `is:public state:open label:help-wanted,helpwanted,help+wanted,help,hacktoberfest updated:>-P1Y` (could save and use whenever)

Arguably, if you considered this proposal, you could get rid of the `<>` and `..` syntaxes in favor of [Range/Period](https://en.wikipedia.org/wiki/ISO_8601#Durations) (`P`) and [Time Interval](https://en.wikipedia.org/wiki/ISO_8601#Time_intervals) `/` syntaxes, possibly simplifying the parsing of the parameters since these are standard ISO capabilities and should be supported by date parsing libraries.  Things then might look like this:

Period: `updated:-P1D`
Interval: `updated:2024-10-12/2024-10-13`

Note: `-P` is via ISO 8601-2:2019(E) ([1](https://github.com/tc39/proposal-temporal/issues/819#issuecomment-673721325), [2](https://www.postgresql.org/message-id/9q0ftb37dv7.fsf%40gmx.us), [Library of Congress Spec (similar)](https://www.loc.gov/standards/datetime/iso-tc154-wg5_n0039_iso_wd_8601-2_2016-02-16.pdf))

**Reply by roke-julian-lockhart** [reply]
@kevinpeno, I agree: I consider ISO 8601-2:2019/Amd 1:2025 to be the sole feature-complete implementation. Do any alternatives, like GitLab and Forgejo, support this?

**Comment by oliverjohns** [comment]
This is a important parameter for github search. Sometimes you want to find new code examples for a new library version/etc, and not have to sift through hundreds of dinosaur results

**Comment by onkoe** [comment]
I'd also like to avoid results after the creation of LLM aggregation repos, which have tons of junk text related to my keywords. It's a particular concern for non-code text, which was an incredible resource on GitHub prior to the popularization of LLMs. :(

For example, I'm currently trying to find information about the Euchee language, but I keep finding lots of aggregation sites instead of actual results. 

Filtering by date would help a lot here! :D

**Comment by tyleralbee** [comment]
I wanted to create a GitHub Projects view that would only show issues created within the last day so that I had an easy way to edit my recently created issues in bulk. 

<img width="1840" height="179" alt="image" src="https://github.com/user-attachments/assets/d6e52891-43d1-4786-96b8-6203850eac37" />
